invenio_communities/communities/permissions.py

```diff
@@ -1,106 +1,107 @@
 #
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Community permissions."""
 
+from elasticsearch_dsl.query import Q
 from flask_principal import UserNeed
-from flask_security import current_user
 from invenio_access.permissions import any_user
-from invenio_records_permissions.generators import AnyUser, Generator
+from invenio_rdm_records.services.generators import \
+    IfRestricted as IfRestrictedBase
+from invenio_records_permissions.generators import AnyUser, \
+    AuthenticatedUser, Generator
 from invenio_records_permissions.policies import BasePermissionPolicy
 
-# from invenio_communities.members.permissions import CommunityMember
 
-
-class CommunityVisibilityPolicy(Generator):
-    """Allows only community members to read the community if its hidden."""
-
-    def needs(self, community=None, **kwargs):
-        """."""
-        if community['visibility'] == 'public':
-            return [any_user]
+class IfRestricted(IfRestrictedBase):
+    """IfRestricted.
+
+    IfRestricted(
+        'record',
+        then_=[RecordPermissionLevel('view')],
+        else_=[ActionNeed(superuser-access)],
+    )
+
+    A record permission level defines an aggregated set of
+    low-level permissions,
+    that grants increasing level of permissions to a record.
+
+    """
+
+    def generators(self, record):
+        """Get the "then" or "else" generators."""
+        if record is None:
+            # TODO - when permissions on links are checked, the record is not
+            # passes properly, causing below ``record.access`` to fail.
+            return self.else_
+        is_restricted = getattr(
+            record.access, self.field, "private")
+        return self.then_ if is_restricted == "private" else self.else_
+
+    def query_filter(self, **kwargs):
+        """Filters for current identity as super user."""
+        q_restricted = Q("match", **{f"access.{self.field}": "private"})
+        q_public = Q("match", **{f"access.{self.field}": "public"})
+        then_query = self.make_query(self.then_, **kwargs)
+        else_query = self.make_query(self.else_, **kwargs)
+        if then_query and else_query:
+            return (q_restricted & then_query) | (q_public & else_query)
+        elif then_query:
+            return (q_restricted & then_query) | q_public
+        elif else_query:
+            return q_public & else_query
         else:
-            return [CommunityMember(['admin', 'curator', 'member'])]
+            return q_public
+
 
+class CommunityOwners(Generator):
+    """Allows community owners."""
 
-class CommunityOwner(Generator):
-    """Returns community owner need."""
+    def needs(self, record=None, **kwargs):
+        """Enabling Needs."""
+        if record:
+            return [
+                UserNeed(owner.owner_id)
+                for owner in record.access.owned_by
+                ]
```

Comment:
```suggestion
            ]
```

Shelved: I see some other places as well with weird indentation. I'm shelving this in a separate issue, so that we merge this guy though.